tls/s2n_handshake_io.c

```diff
@@ -441,6 +441,58 @@ int s2n_conn_set_handshake_no_client_cert(struct s2n_connection *conn) {
     return 0;
 }
 
+int s2n_connection_get_handshake_type(struct s2n_connection *conn) 
+{
+    notnull_check(conn);
+    return conn->handshake.handshake_type;
+}
+
+const char *s2n_translate_handshake_type(int handshake_type)
+{
+    if (handshake_type < 0) {
+        return NULL;
+    }
+
+    if (0 == handshake_type) {
+        return "INITIAL";
+    }
+
+    static char handshake_type_str[MAX_HANDSHAKE_TYPE_LEN];
```

Comment:
To avoid side-effects of calling this function twice and keeping the API simple, how about storing each of the string individually and either lazy-computing them or pre-computing them on initialization? Something like this:

```c
static char handshake_type_str[128][MAX_HANDSHAKE_TYPE_LEN] = {0};

if (handshake_type_str[handshake_type][0] != '\0') {
  return handshake_type_str[handshake_type];
}

/* compute handshake_type_str[handshake_type] */
```

Reply:
Sure.